service/collector/elb.go

```diff
@@ -139,31 +143,77 @@ func (e *ELB) collectForAccount(ch chan<- prometheus.Metric, awsClients clientaw
 
 	var lbs []loadBalancer
 	{
-		i := &elb.DescribeTagsInput{}
-		for _, l := range loadBalancers {
-			i.LoadBalancerNames = append(i.LoadBalancerNames, l.LoadBalancerName)
+		// AWS API has a limit for maximum number of LoadBalancerNames in
+		// single Describe request so it must be done in batches of
+		// maxELBsInOneBatch. In order to not spend so much time on this,
+		// perform requests concurrently and synchronize them with errgroup.
+		errGroup, _ := errgroup.WithContext(context.Background())
+		// Allocate list for ELB tag description results. Length is number of
+		// AWS API batches + 1 for possible left over from uneven number.
+		tagOutputs := make([]*elb.DescribeTagsOutput, len(loadBalancers)/maxELBsInOneDescribeTagsBatch+1)
+
+		i := 0
+		tagInput := &elb.DescribeTagsInput{}
+		for _, lb := range loadBalancers {
+			if len(tagInput.LoadBalancerNames) == maxELBsInOneDescribeTagsBatch {
+				copyTagInput := tagInput
+				errGroup.Go(func() error {
```

Comment:
`i` should be copied here. I guess `DescribeTags` will take so much time that all of the outputs will land in the last index.